lib/membrane_rtsp/transport/tcp_socket.ex

```diff
@@ -52,16 +52,23 @@ defmodule Membrane.RTSP.Transport.TCPSocket do
   @impl true
   def close(_state), do: :ok
 
-  defp recv(acc \\ <<>>) do
-    receive do
-      {:tcp, _socket, data} ->
-        recv(acc <> data)
+  defp recv(socket) do
+    case do_recv(socket) do
+      {:ok, data} ->
+        case Membrane.RTSP.Response.verify_content_length(data) do
```

Comment:
In case the response is partitioned, that function won't ever be called with the "positive" result. I think it would be a pity not to take advantage of the possibility to check the content length after the body is received since the function is already written.

Reply:
I've changed it to run verification until it returns a positive result. Also it should be able to handle all types of troubles: incomplete header, only header, header and part of body.